var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -361,6 +361,8 @@ def codesign_check(self):
 
     def setup_build_environment(self, env):
         env.append_flags("CXXFLAGS", self.compiler.cxx11_flag)
+        if self.spec.satisfies('%intel'):
+            env.set('LDFLAGS', "-shared-intel")
```

Comment:
Will this completely overwrite or append to`LDFLAGS`?

Reply:
Overwrite I'm pretty sure, if that's fixed I'm good with this.  Not terribly fond of this as a solution, but I haven't seen a better option either.

Reply:
I think the cleaner solution is through https://spack.readthedocs.io/en/latest/packaging_guide.html#compiler-flags `flag_handler`.

It's a bit of an obscure function, but you can grep for examples in other packages.

Reply:
@haampie I switched to flag_handler to add -shared-intel in paraview and llvm. Is this what you're thinking of?